R/guides-.r

```diff
@@ -148,6 +154,11 @@ legend_position <- function(position) {
   }
 }
 
+# resolve the guide from the scale and guides
+resolve_guide <- function(aesthetic, scale, guides, default = "none", null = "none") {
+  guides[[aesthetic]] %||% scale$guide %|W|% default %||% null
+}
+
 # validate guide object
 validate_guide <- function(guide) {
```

Comment:
This function does not seem to use the same lookup as the rest of ggplot2 when resolving a string to a function (ie it uses `match.fun()` instead of `find_global()`). Unless there is a compelling reason for this it should use the same search mechanism (this may mean that the plot environment has to be passed down to this function)

Reply:
Can you add a comment to the code so we don't forget